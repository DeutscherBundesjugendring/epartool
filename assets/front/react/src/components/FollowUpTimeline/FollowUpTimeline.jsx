import React from 'react';


const FollowUpTimeLine = props => (
  <div className="well well-bordered well-accent">

    <div className="row offset-bottom-large">
      <div className="col-sm-4">
        <a
          href={props.infoLink}
          title={props.infoLinkTitle}
          className="followup-back-link followup-link media link-unstyled link-unstyled-alt"
        >
          <div className="media-left">
            <div
              className={
                'followup-back-link-question-mark followup-sprite followup-sprite-question-mark'
              }
              aria-hidden="true"
            >
              ?
            </div>
          </div>
          <div className="media-body">
            <strong className="small">
              {props.infoText}
            </strong>
          </div>
        </a>

      </div>
    </div>

    <div className="followup">
      <div className="row">
        <div className="col-sm-4">
          {props.leftColumn && props.leftColumn.map((element, index) => (
            <div key={index}>
              {element}
            </div>
          ))}
        </div>
        <div className="col-sm-4">
          {props.centerColumn && props.centerColumn.map((element, index) => (
            <div key={index}>
              {element}
            </div>
          ))}
        </div>
        <div className="col-sm-4">
          {props.rightColumn && props.rightColumn.map((element, index) => (
            <div key={index}>
              {element}
            </div>
          ))}
        </div>
      </div>
    </div>

  </div>
);

FollowUpTimeLine.propTypes = {
  infoLink: React.PropTypes.string.isRequired,
  infoLinkTitle: React.PropTypes.string.isRequired,
  infoText: React.PropTypes.string.isRequired,
  leftColumn: React.PropTypes.arrayOf(React.PropTypes.element),
  centerColumn: React.PropTypes.arrayOf(React.PropTypes.element),
  rightColumn: React.PropTypes.arrayOf(React.PropTypes.element),
};

export default FollowUpTimeLine;
