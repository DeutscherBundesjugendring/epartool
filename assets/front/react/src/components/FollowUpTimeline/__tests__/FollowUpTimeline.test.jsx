import { shallow } from 'enzyme';
import React from 'react';
import FollowUpTimeline from '../FollowUpTimeline';

describe('rendering', () => {
  it('renders correctly without elements', () => {
    const tree = shallow(
      <FollowUpTimeline
        infoLink="#"
        infoLinkTitle="Link"
        infoText="Text"
        isOpened={() => {}}
      />,
    );

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with element in one column and one row', () => {
    const tree = shallow(
      <FollowUpTimeline
        infoLink="#"
        infoLinkTitle="Link"
        infoText="Text"
        isOpened={() => {}}
        rows={[[<div>Left column element</div>]]}
      />,
    );

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with element in one column and one row and with modal', () => {
    const tree = shallow(
      <FollowUpTimeline
        infoLink="#"
        infoLinkTitle="Link"
        infoText="Text"
        isOpened={() => {}}
        rows={[[<div>Left column element</div>]]}
        modal={<div>Modal element</div>}
      />,
    );

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with elements in more columns and more rows', () => {
    const tree = shallow(
      <FollowUpTimeline
        infoLink="#"
        infoLinkTitle="Link"
        infoText="Text"
        isOpened={() => {}}
        rows={[
          [
            <div>Left column element</div>,
            <div>Center column element</div>,
            <div>Right column element</div>,
          ],
          [
            <div>Left column element</div>,
            <div>Center column element</div>,
          ],
          [
            <div>Left column element</div>,
          ],
        ]}
      />,
    );

    expect(tree).toMatchSnapshot();
  });
});
