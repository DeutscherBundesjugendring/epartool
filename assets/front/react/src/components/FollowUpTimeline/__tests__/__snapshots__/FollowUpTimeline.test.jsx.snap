// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering renders correctly with element in one column and one row 1`] = `
<div
  className="
      well
      well-bordered
      well-accent
      followup-stage
      undefined
    "
>
  <div
    className="row offset-bottom-large"
  >
    <div
      className="col-sm-4"
    >
      <a
        className="followup-back-link followup-link media link-unstyled link-unstyled-alt"
        href="#"
        title="Link"
      >
        <div
          className="media-left"
        >
          <div
            aria-hidden="true"
            className="followup-back-link-question-mark followup-sprite followup-sprite-question-mark"
          >
            ?
          </div>
        </div>
        <div
          className="media-body"
        >
          <strong
            className="small"
          >
            Text
          </strong>
        </div>
      </a>
    </div>
  </div>
  <div
    className="followup"
  >
    <div
      className="followup-rows-container"
    >
      <div
        className="followup-row"
        id="row0"
      >
        <div
          className="followup-row-element"
        >
          <div>
            Left column element
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`rendering renders correctly with element in one column and one row and with modal 1`] = `
<div
  className="
      well
      well-bordered
      well-accent
      followup-stage
      undefined
    "
>
  <div
    className="row offset-bottom-large"
  >
    <div
      className="col-sm-4"
    >
      <a
        className="followup-back-link followup-link media link-unstyled link-unstyled-alt"
        href="#"
        title="Link"
      >
        <div
          className="media-left"
        >
          <div
            aria-hidden="true"
            className="followup-back-link-question-mark followup-sprite followup-sprite-question-mark"
          >
            ?
          </div>
        </div>
        <div
          className="media-body"
        >
          <strong
            className="small"
          >
            Text
          </strong>
        </div>
      </a>
    </div>
  </div>
  <div
    className="followup"
  >
    <div
      className="followup-rows-container"
    >
      <div
        className="followup-row"
        id="row0"
      >
        <div
          className="followup-row-element"
        >
          <div>
            Left column element
          </div>
        </div>
      </div>
    </div>
  </div>
  <div>
    Modal element
  </div>
</div>
`;

exports[`rendering renders correctly with elements in more columns and more rows 1`] = `
<div
  className="
      well
      well-bordered
      well-accent
      followup-stage
      undefined
    "
>
  <div
    className="row offset-bottom-large"
  >
    <div
      className="col-sm-4"
    >
      <a
        className="followup-back-link followup-link media link-unstyled link-unstyled-alt"
        href="#"
        title="Link"
      >
        <div
          className="media-left"
        >
          <div
            aria-hidden="true"
            className="followup-back-link-question-mark followup-sprite followup-sprite-question-mark"
          >
            ?
          </div>
        </div>
        <div
          className="media-body"
        >
          <strong
            className="small"
          >
            Text
          </strong>
        </div>
      </a>
    </div>
  </div>
  <div
    className="followup"
  >
    <div
      className="followup-rows-container"
    >
      <div
        className="followup-row"
        id="row0"
      >
        <div
          className="followup-row-element"
        >
          <div>
            Left column element
          </div>
        </div>
        <div
          className="followup-row-element"
        >
          <div>
            Center column element
          </div>
        </div>
        <div
          className="followup-row-element"
        >
          <div>
            Right column element
          </div>
        </div>
      </div>
      <div
        className="followup-row"
        id="row1"
      >
        <div
          className="followup-row-element"
        >
          <div>
            Left column element
          </div>
        </div>
        <div
          className="followup-row-element"
        >
          <div>
            Center column element
          </div>
        </div>
      </div>
      <div
        className="followup-row"
        id="row2"
      >
        <div
          className="followup-row-element"
        >
          <div>
            Left column element
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`rendering renders correctly without elements 1`] = `
<div
  className="
      well
      well-bordered
      well-accent
      followup-stage
      undefined
    "
>
  <div
    className="row offset-bottom-large"
  >
    <div
      className="col-sm-4"
    >
      <a
        className="followup-back-link followup-link media link-unstyled link-unstyled-alt"
        href="#"
        title="Link"
      >
        <div
          className="media-left"
        >
          <div
            aria-hidden="true"
            className="followup-back-link-question-mark followup-sprite followup-sprite-question-mark"
          >
            ?
          </div>
        </div>
        <div
          className="media-body"
        >
          <strong
            className="small"
          >
            Text
          </strong>
        </div>
      </a>
    </div>
  </div>
  <div
    className="followup"
  >
    <div
      className="followup-rows-container"
    />
  </div>
</div>
`;
